### ISSUE TITLE
Could not determine default i2c bus for platform

### ISSUE THREAD

**Issue by AlexanderKaschta** [PhyPiDAQ/PhyPiDAQ#35]
Currently a `.whl`-file is used for the ADS1115 driver. This (old) python package has Python 2 and Python 3 support, when it was written, and is using the `Adafruit_GPIO` package in the backend. The `Adafruit_GPIO` package is now deprecated and is throwing

> Could not determine default i2c bus for platform

errors, as their included platform-detect code was only written for Raspberry Pi up to the Model 3. Newer revisions, like the Raspi 4 or the Raspi 400 won't get detected and therefore the error is thrown. A fix (could) be simply editing this file, fixing the detection and rebuilding the driver, but that would be not elegant and would create a lot of testing needs.

It would be much more suitable, the migrate now to the new CircuitPython driver from Adafruit, see https://github.com/adafruit/Adafruit_CircuitPython_ADS1x15. It not only includes a up to date driver, but it also drops the Python 2 support (which we don't support anymore) and can be shipped thru `pip`.

**Comment by GuenterQuast** [comment]
yes indeed, full support for this proposal... needs careful testing, of course.

**Comment by AlexanderKaschta** [comment]
The own implementation for the MCP3008 seems to also be affected.

**Comment by AlexanderKaschta** [comment]
Other stuff:

- https://learn.adafruit.com/adafruit-vl53l1x/python-circuitpython
- https://learn.adafruit.com/adafruit-vl53l0x-micro-lidar-distance-sensor-breakout/python-circuitpython
